pkg/controller/nodeipam/ipam/multicidrset/multi_cidr_set_test.go

```diff
@@ -0,0 +1,874 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multicidrset
+
+import (
+	"net"
+	"reflect"
+	"testing"
+
+	"k8s.io/component-base/metrics/testutil"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+func allocateNext(s *MultiCIDRSet) (*net.IPNet, error) {
+	candidate, _, err := s.NextCandidate()
+	if err != nil {
+		return nil, err
+	}
+
+	_ = s.Occupy(candidate)
```

Comment:
why swallow it?

Reply:
I see https://github.com/kubernetes/kubernetes/pull/109090/commits/7c3f82e96bd42ca1038722b5366e1442f0c22a78#r934787147

Reply:
between the call to s.NextCandidate() and s.Occupy() somebody can allocate the candidate

Reply:
+1 this seems to be is a race condition. We need to guard the entire thing:

1. find free range
2. mark range as used

Reply:
Returning the error here, however the race condition is guarded in the code: https://github.com/kubernetes/kubernetes/pull/109090#discussion_r934990045